app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,42 @@
+class PostsController < ApplicationController
+  def index
+    @user = User.find(params[:user_id])
+  end
+
+  def show
+    @user = User.find(params[:user_id])
+    @post = Post.find(params[:id])
+    @like_user = User.first
+  end
+
+  def create
+    @post = Post.new(post_params)
+    @post.author_id = current_user.id
+    @post.likes_counter = 0
+    @post.comments_counter = 0
+    return unless @post.save
+
+    redirect_to user_posts_path
+  end
+
+  def new
+    @post = Post.new
+  end
+
+  def create_comment
+    @comment = Comment.new(comment_params)
+    @comment.post = Post.find(params[:id])
+    @comment.author_id = current_user.id
+    redirect_to user_post_path if @comment.save
+  end
+
```

Comment:
- [ ] @Fahim-Khan-P Kindly remove this method, its good practice to ensure the controller only has methods for a particular view, you already have this method in the comments controller

Reply:
Thank you for your feedback! I agree that it's a duplicated block, I have removed it.